test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+               Throw "Exception expected, execution should not have reached here"
+           }
+           catch
+           {
+                $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'InvalidCastConstructorException'
+           }
+       }
+       
+       
+    Context 'Conversion of Ordered hashtable to PSCustomObject should succeed' {
+      
+       { $script:x = [pscustomobject][ordered]@{one=1;two=2} } | Should Not Throw
+       It '$x is not $null' { $script:x | Should Not Be $null }
+       It '$x type' { $script:x | should BeOfType "System.Management.automation.psobject" }
+       
+       $p = 0
+       # Checks if the first property is One
+       $script:x.psobject.Properties | foreach-object  `
+                                {               
+                                    if ($p -eq 0)  
+                                    {               
+                                        $p++; 
+                                        It 'Name' { $_.Name | Should Be 'one' }
+                                     }
+                                }
+    }
+       
+
+    Context 'Creating an object of an existing type from hashtable should succeed' {       
+       {
+                $source = @"
+                        public class SampleClass1
+                        {                                                       
+                            public SampleClass1(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass1()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $script:x = [SampleClass1]@{a=10;b=5}            
+       } | Should Not Throw
+       It '$x is not $null' { $script:x | Should Not Be $null }
+       It '$x.a' { $script:x.a | Should Be '10' }
+   }
+       
+
+    It 'Creating an object of an existing type from hashtable should throw error when setting non-existent properties' {
+      
+       try
+       { 
+          $source = @"
+                        public class SampleClass2
+                        {
+                            public SampleClass2(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass2()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass2]@{blah=10;b=5 }

```

Comment:
fixed

---
In reply to: [78255577](https://github.com/Powershell/Powershell/pull/2160#discussion_r78255577) [](ancestors = 78255577)